### ISSUE TITLE
tokenize failing on valid python 3.6 code (W605-related)

### ISSUE THREAD

**Issue by delandtj** [hhatto/autopep8#414]
the following code when parsed through autopep8 throws an exception:
```
co = re.compile(
            "\(")
```

```
  File "/home/src/autopep8/autopep8.py", line 3993, in generate_tokens
    tokenize.generate_tokens(string_io.readline)
  File "/usr/lib/python3.6/tokenize.py", line 596, in _tokenize
    raise TokenError("EOF in multi-line statement", (lnum, 0))
tokenize.TokenError: ('EOF in multi-line statement', (2, 0))
```

when the bracket inside the quotes is REMOVED, it is accepted.

when the backslash inside the quotes is REMOVED, it is accepted.

when the string is placed on the SAME LINE as the re.compile, it is accepted.

ONLY when these three EXACT conditions combine, tokenize.py will raise an exception.

**Comment by delandtj** [comment]
raised as bugreport https://bugs.python.org/issue34428

**Comment by delandtj** [comment]
further investigation shows that generate_tokens is being called with single-lines of text.  this affects the state engine behind tokenize.generate_tokens, preventing it from properly detecting line breaks.

if yield were utilised such that the tokenize.generate_tokens could continue with multiple lines it would probably be ok.

**Comment by hhatto** [comment]
Thanks for reporting.

I can't reproduce.
Could you please provide a minimal script that reproduces your issue?
And please provide more information.

* autopep8 version
* Pyhton version
* OS (Linux, Windows, macOS, etc...)

```
$ cat example.py
import re

co = re.compile(
            "\(")
$ autopep8 -d example.py
--- original//Users/hattori/ddd.py
+++ fixed//Users/hattori/ddd.py
@@ -1,4 +1,4 @@
 import re

 co = re.compile(
-            "\(")
+    "\(")
```

**Comment by delandtj** [comment]
cat > x.py << EOF
co = re.compile(
            "\(")
EOF
autopep8 -a -a -a --experimental x.py

**Comment by cclauss** [comment]
Does same happen on Python 3.7?  3.5?

**Comment by hhatto** [comment]
```
$ cat > x.py << EOF
(
"(")
EOF
autopep8 -a -a -a --experimental x.py
```

Umm, I can't reproduce with Python3.5.5, 3.6.6, 3.7.0.

**Comment by hhatto** [comment]
Sorry, I'm mistake.

This is same as #403.
Changed not to raise an exception in 303439802dc194cd1948af33391d59468770acf9.
It will be fixed this PR(PyCQA/pycodestyle/pull/747) is released.
